use-cases/inferencing/cost-optimization/gcsfuse/README.md

```diff
@@ -0,0 +1,271 @@
+without any fine tuning to download the model from the bucket and start
+inference. In the second run, the model will be stored in a hierarchical GCS
+bucket and you will fine-tune GCSFuse configurations to download the model from
+the bucket and start inference.
+
+> Note : By default, a GCS bucket is created as flat.
+
+The goal of this guide is to demonstrate performance improvement in the model
+load time and pod startup time when using fine-tuned configurations with
+GCSFuse.
+
+## Prerequisites
+
+- This guide was developed to be run on the
+  [playground AI/ML platform](/platforms/gke-aiml/playground/README.md). If you
+  are using a different environment the scripts and manifest will need to be
+  modified for that environment.
+
+- Follow [these instructions](/use-cases/prerequisites/storage-benchmarking.md)
+  to download the Llama-3.3-70B-Instruct model into GCS bucket.
+
+## Preparation
+
+- Clone the repository.
+
+  ```sh
+  git clone https://github.com/GoogleCloudPlatform/accelerated-platforms && \
+  cd accelerated-platforms
+  ```
+
+- Change directory to the guide directory.
+
+  ```sh
+  cd use-cases/inferencing/cost-optimization/storage-benchmarking/gcsfuse
+  ```
+
+- Ensure that your `MLP_ENVIRONMENT_FILE` is configured.
+
+  ```sh
+  cat ${MLP_ENVIRONMENT_FILE} && \
+  source ${MLP_ENVIRONMENT_FILE}
+  ```
+
+  > You should see the various variables populated with the information specific
+  > to your environment.
+
+- Get credentials for the GKE cluster.
+
+  ```sh
+  gcloud container fleet memberships get-credentials ${MLP_CLUSTER_NAME} --project ${MLP_PROJECT_ID}
+  ```
+
+- Configure the environment.
+
+  | Variable      | Description                                                                    | Example                |
+  | ------------- | ------------------------------------------------------------------------------ | ---------------------- |
+  | ACCELERATOR   | Type of GPU accelerator to use (a100, h100, l4)                                | a100                   |
+  | MODEL_NAME    | The name of the model folder in the root of the GCS model bucket               | meta-llama             |
+  | MODEL_VERSION | The name of the version folder inside the model folder of the GCS model bucket | Llama-3.3-70B-Instruct |
+
+  ```sh
+  ACCELERATOR="a100"
+  MODEL_NAME="meta-llama"
+  MODEL_VERSION="Llama-3.3-70B-Instruct"
+  ```
+
+## Serve the model with vLLM with no tuning
+
+- Configure the deployment.
+
+  ```
+  VLLM_IMAGE_NAME="vllm/vllm-openai:v0.6.6.post1"
+  ```
+
+  ```sh
+  git restore manifests/model-deployment-${ACCELERATOR}-dws.yaml
+  sed \
+  -i -e "s|V_MODEL_BUCKET|${MLP_STORAGE_BENCHMARK_FLAT_BUCKET}|" \
+  -i -e "s|V_MODEL_NAME|${MODEL_NAME}|" \
+  -i -e "s|V_MODEL_VERSION|${MODEL_VERSION}|" \
+  -i -e "s|V_IMAGE_NAME|${VLLM_IMAGE_NAME}|" \
+  -i -e "s|V_KSA|${MLP_STORAGE_BENCHMARKING_KSA}|" \
+  manifests/model-deployment-${ACCELERATOR}-dws.yaml
```

Comment:
Switch this to use `envsubst` similar to the [RAG guides](https://github.com/GoogleCloudPlatform/accelerated-platforms/blob/main/use-cases/rag-pipeline/embedding-models/multimodal-embedding/README.md)

Reply:
just so I understand it completely, the placeholders like V_MODEL_BUCKET should be directly replaced with the env variables(e.g ${MLP_STORAGE_BENCHMARK_FLAT_BUCKET}) in the manifest?

Reply:
btw, I just tested and found that envsubst doesn't respect `set -o nounset` . 

The following command replaces the environment variable that is not set to blank instead of throwing the `bash: <ENVIRONMENT_VARIABLE> unbound variable error`.

envsubst < manifests/model-deployment-${ACCELERATOR}-dws.yaml| sponge manifests/model-deployment-${ACCELERATOR}-dws.yaml

Reply:
Yeah, that was why we had added it into the `Configure the deployment` step. Ideally we should be setting all of the required variables in that step even if they come directly from the environment configuration file.